app/utils.py

```diff
@@ -154,29 +162,141 @@ def setup_logger(name, level=logging.INFO):
     return logger
 
 
-def validate_resource(json):
-    validation_errors = {'errors': {'type': 'validation'}}
+def validate_resource(request, id=-1):
+    json = request.get_json()
+    validation_errors = {'errors': [],
+                         'type': 'validation'}
+
     if not json:
-        message = "A JSON body is required to use this endpoint, but none was given"
-        validation_errors['errors']['message'] = message
+        message = generate_error(ValidationError.MissingJSONError, field=None)
+        validation_errors['errors'].append(message)
         return validation_errors
 
-    validation_errors['errors']['missing_params'] = []
-
     required = []
     for column in Resource.__table__.columns:
-        if column.nullable is False and column.name != 'id':
-            # strip _id from category_id
-            name = column.name.replace('_id', '')
-            required.append(name)
+
+        # There are only required parameters for POSTing new resources.
+        if request.method == 'POST':
+            if column.nullable is False and column.name != 'id':
+                # strip _id from category_id
+                name = column.name.replace('_id', '')
+                required.append(name)
+
+        # Type Validation.
+        col_type = column.type.python_type
+        field = json.get(column.name.replace('_id', ''))
+
+        if field is not None:
+
+            # Category is a foreign key (int) relation but we only accept string input.
+            # The provided category will get mapped to this resource later on.
+            if column.name == 'category_id':
+                col_type = str
+
+            # Allow String columns to accept integers.
+            if col_type is str:
+                if type(field) in [int, float]:
+
+                    # Do not allow integers as urls
+                    if column.name.replace('_id', '') != 'url':
```

Comment:
We already declare `name` to be `column.name.replace('_id', '')` up on line 182. We keep calling this replace function over and over again, but can we just use `name` instead?

Reply:
Might be good to rename `name` as `column_name` or `param_name` or something because maybe `name` is too generic. But that's not a huge deal, we can leave it as `name` for now